### DISCUSSION TITLE
Change GPU

### DISCUSSION THREAD

**Discussion by Bariatric-surgery** [Project-MONAI/MONAILabel#1177] [General]
I get an error that I am running out of memory while running monailabel radiology app on my server.

We have multiple GPUs and when I checked the GPU: 0 (which got selected by default of monailabel) that it has almost no memory left.

How can I change the GPU which is being used in the code? 
So for example switiching from "GPU: 0" to "GPU: 2" 

Thank you very much!!

**Comment by SachidanandAlle** [comment]
https://stackoverflow.com/questions/39649102/how-do-i-select-which-gpu-to-run-a-job-on

**Comment by SachidanandAlle** [comment]
if you are running training.. MONAILabel supports multi-gpu and gpus flag.. where for you can say `gpus=1,2` to pick others

**Reply by Bariatric-surgery** [reply]
Is there a way to specify the GPU used in the monailabel code itself? so not on environmental variable, but on the python code.

**Comment by SachidanandAlle** [comment]
https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/segmentation.py#L87

here you can pass additional configs
config={"largest_cc": True, "device": "cuda-1"}

For Training, you can pass gpus=1,2 in the train request json. If you see
the 3D slicer, they are already available for user to pass on.
Enable Developer Mode under Edit => Application Settings => MONAI Label =>
Developer Mode

On Mon, Dec 5, 2022 at 11:47 PM Bariatric-surgery ***@***.***>
wrote:

> Is there a way to specify the GPU used in the monailabel code itself? so
> not on environmental variable, but on the python code.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Project-MONAI/MONAILabel/discussions/1177#discussioncomment-4320503>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABX7YK7F24KWTOJZOBEMANDWL3VRPANCNFSM6AAAAAASUK37TE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 

*Thanks & Regards*
*SACHI*

**Reply by Bariatric-surgery** [reply]
Many thanks
